fdbrpc/FlowTransport.actor.cpp

```diff
@@ -737,6 +737,13 @@ ACTOR Future<Void> connectionKeeper(Reference<Peer> self,
 
 				conn->close();
 				conn = Reference<IConnection>();
+
+				// Old versions will throw this error, and we don't want to forget their protocol versions.
```

Comment:
If I understood correctly it is supposed to be old version will not throw this error, and will hang up? https://github.com/apple/foundationdb/blob/master/fdbrpc/FlowTransport.actor.cpp#L1224

Reply:
What happens depends on what side of the connection you are on. If you are sending the connect packet, the connection will drop and you won't learn anything about the protocol version.

If you are receiving the connect packet, then it will hang up on the old connection (i.e. not send anything), but locally it will throw an error that cleans up the connection state. We do, however, learn what the protocol version is in this case, and we're choosing not to discard it by ignoring this error.

Reply:
This differs from the newer versions, by the way, in the fact that new versions leave the connection open but don't exchange any real messages over it. The idea is that by leaving the connection open, we can cheaply keep track of the fact that the two sides are not compatible.